spec/controllers/session_controller_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe Devise::SessionsController, type: :controller do
+  it 'Get to signin page' do
+    @request.env['devise.mapping'] = Devise.mappings[:user]
+    get :new
+    expect(response.status).to eq(200)
+  end
+
+  it 'Get 302 on User and Password request for TOTP' do
```

Comment:
could the test name be a more descriptive? to actually say what it's testing? e.g. `initiates auth responses when no session` and similar with the other e.g. `responds to MFA challenge when MFA present` ?